ironfish/src/blockchain/blockchain.ts

```diff
@@ -985,22 +976,19 @@ export class Blockchain {
       }
 
       const blockNotes = []
-      const blockNullifiers = []
       for (const transaction of transactions) {
         for (const note of transaction.notes()) {
           blockNotes.push(note)
         }
-        for (const spend of transaction.spends()) {
-          blockNullifiers.push(spend.nullifier)
-        }
       }
 
+      const originalNullifierSize = await this.nullifiers.size(tx)
       await this.notes.addBatch(blockNotes, tx)
-      await this.nullifiers.addBatch(blockNullifiers, tx)
 
       const noteCommitment = await this.notes.rootHash(tx)
       const noteSize = await this.notes.size(tx)
-      const nullifierSize = await this.nullifiers.size(tx)
+      const addedNullifiers = transactions.reduce((count, t) => count + t.spendsLength(), 0)
```

Comment:
why add this separate reduce? Just calculate this in the loop above as the count of nullifiers in the block.

Reply:
Yea, alright changed that